underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystemFactory.java

```diff
@@ -101,6 +101,12 @@ private boolean addAndCheckSwiftCredentials() {
       Configuration.set(passwordKeyConf, System.getProperty(passwordKeyConf));
     }
 
+    String swiftSimulationConf = Constants.SWIFT_SIMULATION;
+    if (System.getProperty(swiftSimulationConf) != null
+        && !Configuration.containsKey(swiftSimulationConf)) {
```

Comment:
is there a default value for `Constants.SWIFT_SIMULATION` in `Configuration` ?
If so, you can skip checking `containsKey`, and instead use `Configuration.get(swiftSimulationConf) == null`